Project/Back-End/user_app/views.py

```diff
@@ -9,43 +9,66 @@
 from rest_framework.status import HTTP_201_CREATED, HTTP_404_NOT_FOUND, HTTP_204_NO_CONTENT, HTTP_200_OK
 from applicant_app.models import Applicant
 from recruiter_app.models import Recruiter
+from company_app.models import Company
+from education_app.models import Education
 
 # Create your views here.
 class Sign_Up(APIView):
     # Signs up users and creates respective account type, Refactor to Groups
     def post(self, request):
         # sets username to email
+        print(request.data)
         request.data["username"] = request.data.get("email")
+        email = request.data.get("email")
+        username = request.data["username"]
+        password = request.data.get("password")
+        name = request.data.get("name")
+        account_type = request.data.get("account_type")
+        company_name = request.data.get("company")
+        education = request.data.get("education")
+        school = request.data.get("school")
+        field = request.data.get("field")
+        if company_name:
+            company = Company.objects.get_or_create(name = company_name)
+        if education:
+             education = Education.objects.get_or_create(degree_type=education,school_name = school, degree_field = field)
         # create new user
-        new_user = User.objects.create_user(**request.data)
+        new_user = User.objects.create_user(username=username,email=email, password=password, name=name, account_type=account_type)
         token = Token.objects.create(user = new_user)
+        print(new_user)
         # create applicant or recruiter based on account type
-        if new_user.account_type.lower() == "applicant":
-            Applicant.objects.create(email = new_user)
-        else:
-            Recruiter.objects.create(email = new_user)
+        if new_user.account_type == "applicant":
+            Applicant.objects.create(email = new_user, education = education[0])
+        elif new_user.account_type == "recruiter":
+            Recruiter.objects.create(email = new_user, company = company[0])
         return Response({"user": new_user.email, "account_type":new_user.account_type,"token":token.key}, status=HTTP_201_CREATED)
 class Log_In(APIView):
     def post(self, request):
          # Authenticates the User an Logs them in
-        user = authenticate(**request.data)
-        if user:
-            token, created = Token.objects.get_or_create(user = user)
-            return Response({"token":token.key,"user":user.email}, status=HTTP_200_OK)
-        return Response("INVALID CREDENTIALS", status=HTTP_404_NOT_FOUND)
+            print(request.data)
+            user = authenticate(**request.data)
+            if user:
+                token, created = Token.objects.get_or_create(user = user)
+                return Response({"token":token.key,"user":user.email}, status=HTTP_200_OK)
+            else:
+                 return Response("USER NOT SIGNED IN",status=HTTP_404_NOT_FOUND)
```

Comment:
A better error status might be HTTP_401_UNAUTHORIZED or HTTP_FORBIDDEN (the former appears more appropriate here). 401s are used generally when credentials are invalid and 403s are used when credentials are valid but the associated account does not have the proper authority.